### PR TITLE
Fix push notification UX: actionable feedback for all error states

### PR DESCRIPTION
**Pull request by KristinnRoach**
## Summary

- **Browser-specific blocked guidance**: When notifications are denied, users now see instructions specific to their browser (Chrome, Firefox, Edge, Safari) instead of a generic "Check browser settings" message
- **Silent-block detection**: When a browser auto-blocks notifications without showing a prompt, a distinct message explains what happened and how to fix it
- **FCM token failure feedback**: Previously, if permission was granted but the FCM token setup failed silently, a false "success" toast was shown. Now shows an error toast with retry option
- **Defensive unsupported guard**: The enable prompt now handles the edge case where `requestPermission()` returns `unsupported`, routing to the push-unsupported notification

## Test plan

- [x] 10 new controller unit tests covering all `requestPermission()` code paths (granted, denied, silent-block, dismissed, enable-failed, callbacks)
- [x] 5 new `getBlockedMessage()` tests (browser-specific keys, generic fallback, silent-block interpolation, undefined browser)
- [x] All 274 tests passing
- [ ] Manual: test enable prompt in Chrome, Firefox, Safari with notifications blocked
- [ ] Manual: test with invalid VAPID key to trigger enable-failed path

## Summary by CodeRabbit

* **New Features**
  * Notifications now show browser-specific blocked messages and contextual warnings (Chrome, Firefox, Safari, Edge)
  * Permission flows provide clearer, contextual feedback for failures and silent-block situations

* **Translations**
  * Added English and Icelandic strings for browser-specific blockage, silent-block, and failure messages

* **Tests**
  * Added comprehensive tests covering permission outcomes, silent-block detection, and messaging behavior

### CODE REVIEW

File: src/components/notifications/enable-notifications-prompt.js [hunk 1]
```diff
@@ -46,14 +66,24 @@ export function showEnableNotificationsPrompt() {
...
+            // Permission granted but FCM token/setup failed
+            showErrorToast(t('notification.enable.failed'));
+            btn.disabled = false;
+            btn.textContent = t('shared.enable');
+          } else if (result.reason === 'unsupported') {
+            // Defensive: shouldn't normally reach here, but handle gracefully
+            import('./push-unsupported-notification.js').then(({ showPushUnsupportedNotification }) => {
+              showPushUnsupportedNotification();
+            });
+            inAppNotificationManager.remove(NOTIFICATION_ID);
```
[the remark this answers is not in the document]
Comment (KristinnRoach) [reply src/components/notifications/enable-notifications-prompt.js]: Is there a good reason for the dynamic import? Why not just static import with the rest at the top of the file? @claude  @copilot
